src/main/java/frc/robot/commands/CoralArmCommand.java

```diff
@@ -1,37 +1,52 @@
 package frc.robot.commands;
 
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
 import edu.wpi.first.wpilibj2.command.Command;
 import frc.robot.subsystems.CoralArm;
 
 public class CoralArmCommand extends Command {
 
     private final CoralArm arm;
+    private final TrapezoidProfile.Constraints constraints;
+
+    private TrapezoidProfile motionProfile;
+    private TrapezoidProfile.State motionProfileGoal;
+    private TrapezoidProfile.State motionProfileSetPoint;
 
     private double targetPositionDegrees;
     private boolean hasNewTarget;
     private boolean didReachPosition;
+    private boolean isHolding;
 
     public CoralArmCommand(CoralArm arm) {
         this.arm = arm;
+        constraints = new TrapezoidProfile.Constraints(1000,200);
 
         addRequirements(arm);
     }
 
     @Override
     public void initialize() {
-
+        isHolding = false;
     }
-
     @Override
     public void execute() {
         if (hasNewTarget) {
             // reset to move to new angle
-
-            didReachPosition = false;
-            hasNewTarget = false;
+            if(!isHolding){
+                arm.stop();
+            }else {
+
+                didReachPosition = false;
+                hasNewTarget = false;
+            }
+            motionProfile = new TrapezoidProfile(constraints);
+            motionProfileGoal = new TrapezoidProfile.State(targetPositionDegrees,0);
+            motionProfileSetPoint = new TrapezoidProfile.State(arm.getPositionDegrees(),0);
         }
-
-        arm.setMoveToPosition(targetPositionDegrees);
+        motionProfileSetPoint = motionProfile.calculate(0.02, motionProfileSetPoint, motionProfileGoal);
```

Comment:
don't use the motion profiling when the arm has reached the target position. Use it until that point, but afterwards stop using it